### PR TITLE
[#3550] Port CloudAdapter & MSI to TypeScript samples (2/2)

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses #3550

## Description
This PR updates the TypeScript samples, adding support for CloudAdapter and MSI.

### Detailed Changes
- Updated the **_botbuilder_** dependency to the RC0 version.
- Updated **_index.js_** to:
   - Replace BotFrameworkAdapter with CloudAdapter.
   - Added an instance of ConfigurationServiceClientCredentialFactory to manage credentials.
   - Pass the new parameters AppType and AppTenantId to the _ConfigurationServiceClientCredentialFactory_ instance.
- Added the new settings _MicrosoftAppType_ and _MicrosoftAppTenantId_ to the **_env_** file.
- Added missing **env** files.
- Updated **TeamsConversationBot** and **TeamsStartNewThreadInChannel** classes to adapt the code to the new adapter's methods.

This PR updates the following samples:
- 16.proactive-messages
- 50.teams-messaging-extensions-search
- 51.teams-messaging-extensions-action
- 57.teams-conversation-bot
- 58.teams-start-new-thread-in-channel

## Testing
The following images show the teams-messaging-extensions-action sample properly configured and working with SingleTenant.
image

### CODE REVIEW

File: samples/typescript_nodejs/16.proactive-messages/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
         "url": "https://github.com"
     },
     "dependencies": {
-        "botbuilder": "~4.14.0",
+        "botbuilder": "~4.15.0-rc0",
```
Comment (BruceHaley) [inline samples/typescript_nodejs/16.proactive-messages/package.json:21]: I would avoid making a dependency on a release candidate, code that will be out of date in a few days if not already.

**Review by BruceHaley — CHANGES_REQUESTED** [review]

File: samples/typescript_nodejs/16.proactive-messages/package.json (hunk 1 above)
Comment (ceciliaavila) [reply samples/typescript_nodejs/16.proactive-messages/package.json:21]: Hi @BruceHaley, we are waiting for the 4.15 release to update all the samples. Meanwhile, we use the RC version to test MSI support.

Comment (mrivera-ms) [inline samples/typescript_nodejs/16.proactive-messages/package.json:21]: >4.15.0-rc0" [](http://example.com/codeflow?start=24&length=11)

4.15.0 has been published.

Comment (ceciliaavila) [reply samples/typescript_nodejs/16.proactive-messages/package.json]: Hi @mrivera-ms, we updated the packages. Thanks!

**Review by mrivera-ms — APPROVED** [review]

**Review by BruceHaley — APPROVED** [review]

**Pull request merged**